Source/WebKit/UIProcess/mac/DisplayLink.cpp

```diff
@@ -110,97 +108,97 @@ void DisplayLink::addObserver(IPC::Connection::UniqueID connectionID, DisplayLin
     }
 }
 
-void DisplayLink::removeObserver(IPC::Connection::UniqueID connectionID, DisplayLinkObserverID observerID)
+void DisplayLink::removeObserver(Client& client, DisplayLinkObserverID observerID)
 {
     ASSERT(RunLoop::isMain());
 
     Locker locker { m_observersLock };
 
-    auto it = m_observers.find(connectionID);
+    auto it = m_observers.find(&client);
     if (it == m_observers.end())
         return;
 
-    auto& connectionInfo = it->value;
+    auto& clientInfo = it->value;
 
-    bool removed = connectionInfo.observers.removeFirstMatching([observerID](const auto& value) {
+    bool removed = clientInfo.observers.removeFirstMatching([observerID](const auto& value) {
         return value.observerID == observerID;
     });
     ASSERT_UNUSED(removed, removed);
 
     LOG_WITH_STREAM(DisplayLink, stream << "[UI ] DisplayLink " << this << " for display " << m_displayID << " remove observer " << observerID);
 
-    removeInfoForConnectionIfPossible(connectionID);
+    removeInfoForClientIfPossible(client);
 
     // We do not stop the display link right away when |m_observers| becomes empty. Instead, we
     // let the display link fire up to |maxFireCountWithoutObservers| times without observers to avoid
     // killing & restarting too many threads when observers gets removed & added in quick succession.
 }
 
-void DisplayLink::removeObservers(IPC::Connection::UniqueID connectionID)
+void DisplayLink::removeClient(Client& client)
 {
     ASSERT(RunLoop::isMain());
 
     Locker locker { m_observersLock };
-    m_observers.remove(connectionID);
+    m_observers.remove(&client);
 
     // We do not stop the display link right away when |m_observers| becomes empty. Instead, we
     // let the display link fire up to |maxFireCountWithoutObservers| times without observers to avoid
     // killing & restarting too many threads when observers gets removed & added in quick succession.
 }
 
-void DisplayLink::removeInfoForConnectionIfPossible(IPC::Connection::UniqueID connectionID)
+void DisplayLink::removeInfoForClientIfPossible(Client& client)
```

Comment:
Maybe we can just use `removeInfoForClient()` and we can return bool to indicate whether the info was removed or not. This is what `HashMap::remove()`.

Reply:
I can do this.